lib/timers.js

```diff
@@ -418,6 +404,27 @@ function processImmediate() {
 }
 
 
+function tryOnImmediate(immediate, queue) {
```

Comment:
Can you move the deopt comment to here and clarify it a tad, ala https://github.com/nodejs/node/pull/4007/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR178 ?